oak/server/rust/oak_runtime/src/runtime/mod.rs

```diff
@@ -630,6 +631,7 @@ impl RuntimeRef {
         entrypoint: &str,
         label: &oak_abi::label::Label,
         reader: Handle,
+        writer: Handle,
```

Comment:
If I add a `writer` handle as a parameter to this function, it means that I will need to pass some `writer` here:
https://github.com/project-oak/oak/blob/589b2d20bbd9e0d8388f58e8b50b50522842013c/oak/server/rust/oak_runtime/src/node/wasm.rs#L198-L211
And IIUC we don't have `None` handles, right?
cc @tiziano88 @daviddrysdale @blaxill

Reply:
I have added `writer` as a parameter to the `node_create`:
https://github.com/project-oak/oak/blob/c2b2ff4b666de2a4f5fe7ff7cbfadbfc00af69f2/oak/server/rust/oak_runtime/src/runtime/mod.rs#L773-L780
But this function is called inside a Wasm node, when `oak_abi` funciton `node_create` is called (and it doesn't provide a `writer`).

Is it ok to search for this `writer` in the list of channels and pass to the `node_create` in the Wasm node, or it may affect the Control Flow?

Reply:
I think perhaps we should have a pre-requisite to this PR to make gRPC node creation more similar to the creation of other pseudo-nodes, i.e. let the wasm node create them. This is what we do for logging, for instance.

Reply:
I already done this in this PR:
https://github.com/project-oak/oak/pull/772/files#diff-ce2a63846754333df981075c4b07ec96R93-R101

Reply:
e.g.:

https://github.com/project-oak/oak/blob/006b9210e02e6483d298d24ec2940cff394906ee/sdk/rust/oak/src/logger/mod.rs#L80-L99

this is what creates a log pseudo-node, and we also have a default config name for it. We could have the same for gRPC.

Reply:
So pseudo-nodes should be created by the Oak Node, right?
But Runtime also creates a pseudo-node (and passes it a reader):
https://github.com/project-oak/oak/pull/772/files#diff-6c581d36a36131023ea8060843bb2dffR126

Reply:
Ah, makes sense, but then I think we should also flip the direction of the channel, so that the gRPC server pseudo-node would have an incoming *read* channel, like every other node, over which the caller passes a single write end of a channel to receive invocations. Would that work?

Reply:
Yes, I think it will work.
So before starting the server, gRPC pseudo-node should wait on the channel to receive the writer.

Reply:
I think the server may as well start as soon as possible, but then at the first incoming gRPC request it would block (`wait_on_channels`) to receive the channel to which to write the invocation, store it somewhere in its local state, and then use that channel for all subsequent invocations.

Reply:
Done
Created an `init` function:
https://github.com/project-oak/oak/pull/772/files#diff-1211ded35ba967a10825322cdf0cbd20R362-R378

It returns a **read** handle to read gRPC invocations from. This handle should be used by the Runtime, but I haven't found a loop/server inside the Runtime that reads invocations from a channel and dispatches them (runs corresponding gRPC method).